File: Python/Queue/PriorityQueue.py
# implementation of priority queue in python

class PriorityQueue(object):
    def __init__(self):
        self.queue = []

    def __str__(self):
        return ' '.join([str(i) for i in self.queue])
    
    def is_empty(self):
        return len(self.queue) == 0

    def join_queue(self, data):
        self.queue.append(data)    
    
    def serve(self):    
        try:
            max = 0
            for i in range(len(self.queue)):
                if self.queue[i] > self.queue[max]:
                    max = i
            
            item = self.queue[max]
            del self.queue[max]
            return item

        except IndexError:
            print()
            exit()


queue = PriorityQueue()
queue.join_queue(12)
queue.join_queue(1)
queue.join_queue(14)
queue.join_queue(7)

print(queue)

while not queue.is_empty():
    print(f"{queue.serve()}", end=", ")

        

File: Javascript/Queue/PriorityQueue.js
// javascript implementation of the priority queue

class PriorityQueue {
    constructor () {
        this.queue = []
    }

    toString () {
        return this.queue.join(' ')
    }

    isEmpty () {
        return this.queue.length === 0
    }

    joinQueue (data) {
        return this.queue.push(data)
    }

    serve () {
        try {
            let maxIndex = 0
            for(let i=0; i < this.queue.length; i++) {
                if (this.queue[i] > this.queue[maxIndex]) {
                    maxIndex = i
                }
                const item = this.queue[maxIndex]
                this.queue.splice(maxIndex, 1)
                return item
            }
        } catch (error) {
            console.log(error);
            process.exit()
        }
    }

}

let myqueue = new PriorityQueue()
myqueue.joinQueue(12)
myqueue.joinQueue(1)
myqueue.joinQueue(14)
myqueue.joinQueue(7)

console.log(myqueue);
